### PR TITLE
Implement largeTitle backgroundColor and hideShadow on iOS

### PR DESCRIPTION
**Pull request by dsolimando**
New props allowing to:
- define the background color 
- hide the shadow

of the header when the edge of any scrollable content reaches the matching edge of the navigation bar.

See [scrollEdgeAppearance](https://developer.apple.com/documentation/uikit/uinavigationitem/3198041-scrolledgeappearance)

### CODE REVIEW

**Comment by kmagiera** [comment]
Thanks for working on this! I'd be interested in learning more about the differences between scrollEdgeShadow and hideShadow props. It is not clear for me from the jsdoc so some clarification there would be appreciated.

**Comment by dsolimando** [comment]
You are welcome!

Basically in the context of large titles in headers, scrollEdgeShadow allows to hide/show shadow **only when large title is displayed** meaning when no scroll happent. 

As soon as user scrolls, the large title disappears and the header height decreases, hideShadow prop is used.

**Comment by kmagiera** [comment]
ok, I think I understand now what it does, however I still think that the description and naming (despite naming coming from the UIKit API) is quite confusing. Would it be sensible to call it `largeTitleHeaderBackgroundColor` and `largeTitleHeaderHideShadow` ? Or they also apply in other cases and are not coupled to the large title setting?

**Comment by dsolimando** [comment]
No it's a good idea!

will do it

**Comment by dsolimando** [comment]
Just did it

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-native-screens",
-  "version": "2.0.0-beta.4",
+  "version": "2.0.0-beta.10",
```
Comment (kmagiera) [inline package.json]: mind reverting this change?

**Comment by dsolimando** [comment]
Hello!

I just merged upstream into my branch to fix conflicts.

It's ready to be merged

**Comment by dsolimando** [comment]
Hello,

any plan to merge the PR ?

Thank you

**Comment by kmagiera** [comment]
Hey! So sorry for the delay, was about to merge this one along a few other PRs but apparently after merging this one older PR #307 there are now new conflicts to be resolved. Should be relatively easy to resolve so if you don't have time I can look in near future but if you do I promise to merge this as a top priority to avoid further conflicts

**Pull request merged**